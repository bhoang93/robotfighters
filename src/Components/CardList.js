import React from 'react';
import Card from '../Components/Card';

const CardList = ({ robots }) => {
	return (
		<div> 
			{
		robots.map((user, i) => {
		return (<Card 
			key={robots[i].id}
			id={robots[i].id} 
			name={robots[i].name} 
			email={robots[i].email}
			 />
			);
	})
	}
			</div>
		);
	} // DIV IS IMPORTANT FFS

export default CardList;