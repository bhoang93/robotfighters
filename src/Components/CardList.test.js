import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';

it('expects to render Card component', () => {
	const mockRobots = [
		{
			id: 1,
			name: 'Chester',
			username: 'Terry',
			email: 'f@1.com'
		}
	]
	expect(shallow(<CardList robots={mockRobots}/>)).toMatchSnapshot();
})
