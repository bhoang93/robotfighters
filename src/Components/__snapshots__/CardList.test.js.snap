// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render Card component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "f@1.com",
              "id": 1,
              "name": "Chester",
              "username": "Terry",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="f@1.com"
          id={1}
          name="Chester"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "f@1.com",
          "id": 1,
          "name": "Chester",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="f@1.com"
            id={1}
            name="Chester"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "f@1.com",
            "id": 1,
            "name": "Chester",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
